### DISCUSSION TITLE
Distance from Earth to other solar system bodies.

### DISCUSSION THREAD

**Discussion by Tichro** [cosinekitty/astronomy#361] [Q&A] [answered]
Have just discovered this library of routines and it is exactly what I have been looking for, very impressive work indeed.
Now to my question,,, how can I get the distances from Earth to other solar system bodies?
I have searched for similar discussions but couldn't find anything.
If this is already a function in the library I apologise.
Any chance of an example, I'm using Java in an Android app?

**Comment by cosinekitty** [comment, marked as answer]
Hi @Tichro and welcome to the project! You can use the `geoVector` function to calculate a 3D vector from the center of the Earth to the center of a planet, the Sun, or the Moon. Then you can use the `length` member function to find the distance expressed in astronomical units:

```java
Vector vec = Astronomy.geoVector(Body.Jupiter, time, Aberration.Corrected);
double dist = vec.length();
```

**Reply by Tichro** [reply]
Hi there @cosinekitty , wasn't expecting such a fast response :)
Oh that is superb, and so simple, thank you so much, greatly appreciated, and thank you for such a great library of routines.
